omod/src/main/java/org/openmrs/module/oauth2login/web/JwtTokenUtils.java

```diff
@@ -0,0 +1,22 @@
+package org.openmrs.module.oauth2login.web;
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jwts;
+import sun.security.rsa.RSAPublicKeyImpl;
+
+import java.security.InvalidKeyException;
+import java.util.Base64;
+
+/**
+ * Utility class for parsing and verifying JWT tokens
+ */
+public class JwtTokenUtils {
+	
+	public static Claims parseAndVerifyToken(String jwtToken, String verificationKey) throws InvalidKeyException {
+		
+		return Jwts.parserBuilder().setSigningKey(RSAPublicKeyImpl.newKey(Base64.getDecoder().decode(verificationKey)))
```

Comment:
It's probably best to use the public Java classes to get the RSAPublicKey rather than the `sun.security.rsa` internals, e.g.,

```java
KeyFactory keyFactory = KeyFactory.getInstance("RSA");
X509EncodedKeySpec keySpec = new X509EncodedKeySpec(Base64.getDecoder().decode(verificationKey));
RSAPublicKey publicKey = keyFactory.generatePublic(keySpec);
```

This produces the same result, but those classes are guaranteed to be there and accessible.